eng/pipelines/test-integration-job.yml

```diff
@@ -44,7 +44,7 @@ steps:
       ArtifactName: '$(System.JobAttempt)-Logs ${{ parameters.configuration }} OOP64_${{ parameters.oop64bit }} OOPCoreClr_${{ parameters.oopCoreClr }} LspEditor_${{ parameters.lspEditor }} $(Build.BuildNumber)'
       publishLocation: Container
     continueOnError: true
-    condition: not(succeeded())
+    condition: failed()
```

Comment:
I believe timeout is a second case where we still want logs

Reply:
Thanks for catching. My intuition was that timeouts would count as failure, but upon more research, I believe you are correct that the status would be cancelled. 

Found in a comment on the following page https://docs.microsoft.com/en-us/azure/devops/pipelines/process/phases?view=azure-devops&tabs=yaml#timeouts
```yaml
timeoutInMinutes: number # how long to run the job before automatically cancelling
```

Closing this PR.